### PR TITLE
Introduce `TrieRefcountChange` for better readability

### PR DESCRIPTION
**Pull request by Longarithm**
One step to solve https://github.com/near/NEPs/issues/217

It took a while to understand what exactly `(CryptoHash, Vec<u8>, u32)` stores, so I make a struct which self-describes it.

### CODE REVIEW

File: core/store/src/trie/mod.rs [hunk 1]
```diff
@@ -683,13 +695,17 @@ impl Trie {
...
         let mut insertions = Vec::new();
         for (key, (value, rc)) in changes.into_iter() {
             match rc.cmp(&0) {
-                Ordering::Greater => insertions.push((key, value, rc as u32)),
-                Ordering::Less => deletions.push((key, value, (-rc) as u32)),
+                Ordering::Greater => {
+                    insertions.push(TrieRefcountChange { key_hash: key, value, rc: rc as u32 })
+                }
+                Ordering::Less => {
+                    deletions.push(TrieRefcountChange { key_hash: key, value, rc: (-rc) as u32 })
```
Comment (bowenwang1996) [inline core/store/src/trie/mod.rs:707]: This feels a bit weird. Why do we want to cast it as u32?

File: core/store/src/trie/state_parts.rs [hunk 2]
```diff
@@ -542,11 +544,11 @@ mod tests {
...
             assert!(changes_set.deletions.is_empty(), "state parts only have insertions");
-            for (key, value, rc) in changes_set.insertions {
-                map.entry(key).or_insert_with(|| (value, 0)).1 += rc as i32;
+            for TrieRefcountChange { key_hash, value, rc } in changes_set.insertions {
+                map.entry(key_hash).or_insert_with(|| (value, 0)).1 += rc as i32;
             }
-            for (key, value, rc) in changes_set.deletions {
-                map.entry(key).or_insert_with(|| (value, 0)).1 -= rc as i32;
+            for TrieRefcountChange { key_hash, value, rc } in changes_set.deletions {
+                map.entry(key_hash).or_insert_with(|| (value, 0)).1 -= rc as i32;
```
Comment (bowenwang1996) [inline core/store/src/trie/state_parts.rs:551]: I am not sure why we need to cast back and forth between u32 and i32.

File: core/store/src/trie/mod.rs (hunk 1 above)
Comment (Longarithm) [reply core/store/src/trie/mod.rs:707]: Here changes are separated to insertions and deletions.
Not sure why it is needed, but for insertion or deletion `refcount` change number is always positive. `u32` seems suitable to emphasize this fact.

Comment (bowenwang1996) [reply core/store/src/trie/mod.rs:707]: > refcount change number is always positive

Not entirely sure what you mean. Do you mean that the result is always nonnegative? That does not change the fact that deletion will decrease rc, which means that it is negative here. Also later on we cast it back to i32 to calculate the resulting rc. So I am not sure why we need this casting here.

**Review by bowenwang1996 — APPROVED** [review]

File: core/store/src/trie/mod.rs (hunk 1 above)
Comment (Longarithm) [reply core/store/src/trie/mod.rs:707]: Discussed offline

**Pull request merged**